src/prpy/base/mico.py

```diff
@@ -99,4 +109,11 @@ def Servo(self, velocities):
             self.GetRobot().GetController().Reset(0)
             self.servo_simulator.SetVelocity(velocities)
         else:
-            raise NotImplementedError('Servo is not implemented.') 
+            self.SendVelocitiesToMico(velocities)
+            #reset watchdog timer
+            self.servo_watchdog.reset()
+
+
+    def SendVelocitiesToMico(self, velocities):
+        for velocity_publisher,velocity in zip(self.velocity_publishers, velocities):
+            velocity_publisher.publish(velocity)
```

Comment:
You should introduce a `Lock` around this loop. This function is called from the user's thread(s) and the watchdog thread. I am not sure if `publish` is threadsafe. Even if it is, bad things will happen if `publish` commands are interleaved between two threads.